torch/nn/modules/normalization.py

```diff
@@ -292,6 +292,73 @@ def extra_repr(self) -> str:
             'affine={affine}'.format(**self.__dict__)
 
 
+class RMSNorm(Module):
+    """Applies Root Mean Square Layer Normalization over a mini-batch of inputs.
+
+    This layer implements the operation as described in
+    the paper `Root Mean Square Layer Normalization <https://arxiv.org/pdf/1910.07467.pdf>`__
+
+    .. math::
+        y = \frac{x}{\\sqrt{\\mathrm{RMS}[x] + \\epsilon}} * \\gamma
+
+    The root mean squared norm is taken over the last `D` dimensions, where `D`
+    is the dimension of :attr:`normalized_shape`. For example, if :attr:`normalized_shape`
+    is ``(3, 5)`` (a 2-dimensional shape), the rms norm is computed over
+    the last 2 dimensions of the input.
+
+    Args:
+        normalized_shape (int or list or torch.Size): input shape from an expected input
+            of size
+
+            .. math::
+                [* \times \text{normalized\\_shape}[0] \times \text{normalized\\_shape}[1]
+                    \times \\ldots \times \text{normalized\\_shape}[-1]]
+
+            If a single integer is used, it is treated as a singleton list, and this module will
+            normalize over the last dimension which is expected to be of that specific size.
+        eps: a value added to the denominator for numerical stability. Default: 1e-6
```

Comment:
We should have eps be dtype-dependent. I think it's an oversight on our older modules where we assume they would always run in fp32.
I think we should start using `torch.finfo(x.dtype).eps` here WDYT?
cc @ngimel